pyomo/contrib/doe/scenario.py

```diff
@@ -61,235 +67,87 @@ def __init__(self, para_dict, formula='central', step=0.001, store=False):
-            self.para_dict, self.scena_keys, self.scena, self.formula, self.step
-        )
-        scenario_overall = scenario_object.create_scenario()
-
-        # store scenario
-        if self.store:
-            with open('scenario_simultaneous.pickle', 'wb') as f:
-                pickle.dump(scenario_overall, f)
+        self.generate_scenario()
 
-        return scenario_overall
-
-    def next_sequential_scenario(self, count):
+    def generate_scenario(self):
         """
-        Generate a single scenario class for one of the sequential models
-
-        Parameters:
-        ----------
-        count: the No. of the sequential models
+        Generate scenario dict
 
         Returns:
         -------
-        scenario_next: scenario dict for this sequential model
-        """
-        scena_keys, scena = self._scena_generate(list(self.scena[count].values()), None)
-
-        # each model is basically a 'none' case of an invasive model
-        scenario_object = Scenario_data(
-            self.scena[count], scena_keys, scena, None, self.step
-        )
-        scenario_next = scenario_object.create_scenario()
-
-        return scenario_next
-
-    def generate_sequential_para(self):
-        """
-        Generate object and some 'parameters' for sequential models
-
-        Returns (added to self object):
-        -------
-        self.scena_keys: scenario name, a list of numbers
-        self.scena: a list of parameter dictionaries for all sequential models
-        self.scenario_para: a list of two No. of models involved in calculating one parameter sensitivity
-        self.eps_abs: keys are parameter name, values are the step it is perturbed
-        """
-
-        scena_keys, scena = self._scena_generate(self.scenario_nominal, self.formula)
-        self.scena_keys = scena_keys
-        self.scena = scena
-
-        # record the number of scenarios involved in calculating a certain parameter sensitivities
-        scenario_para = {}
-        for p, para in enumerate(self.para_names):
-            # the scenario involved in Jacobian calculation
-            if self.formula == 'central':
-                scenario_para[para] = [p, p + self.no_para]
-            elif self.formula == None:
-                raise ValueError('Finite difference scheme should be chosen.')
-            else:
-                scenario_para[para] = [p, self.no_para]
-
-        self.scenario_para = scenario_para
-
-        # calculate the perturbation size of every parameter
-        eps_abs = {}
-        for para in self.para_names:
-            # for central difference scheme, perturbation size is two times the step size
-            eps_abs[para] = self.step * self.para_dict[para]
-            if self.formula == 'central':
-                eps_abs[para] *= 2
+        scenario_dict: a dictionary containing scenarios dictionaries.
+        scenario_dict['sceanrio']: a list of dictionaries, each dictionary contains a perturbed scenario
```

Comment:
scenario_dict: a dictionary mapping {scenario —> a list of …}

Reply:
Nevermind, @jsiirola says there is not a great way to document this.

Reply:
Instead, before defining this class, add `ScenarioData = namedtuple(‘ScenarioData’, [‘scenario’, ‘scenario_num’, ‘eps-ads’]`

Reply:
See https://www.geeksforgeeks.org/namedtuple-in-python/#

Reply:
scenario_data.number or scenario_data.name, or scenario_data.eps

Reply:
Changed scenario dictionary to be named tuples throughout code.